src/Orleans.Core/Streams/Internal/StreamSubscriptionHandleImpl.cs

```diff
@@ -84,58 +94,68 @@ public async Task<StreamHandshakeToken> DeliverBatch(IBatchContainer batch, Stre
             }
             else
             {
-                foreach (var itemTuple in batch.GetEvents<T>())
+                if (this.observer != null)
+                {
+                    foreach (var itemTuple in batch.GetEvents<T>())
+                    {
+                        await NextItem(itemTuple.Item1, itemTuple.Item2);
+                    }
+                } else
                 {
-                    await NextItem(itemTuple.Item1, itemTuple.Item2);
+                    await NextItems(batch.GetEvents<T>());
                 }
             }
 
             if (IsRewindable)
             {
-                expectedToken = StreamHandshakeToken.CreateDeliveyToken(batch.SequenceToken);
+                this.expectedToken = StreamHandshakeToken.CreateDeliveyToken(batch.SequenceToken);
             }
             return null;
         }
 
         public async Task<StreamHandshakeToken> DeliverItem(object item, StreamSequenceToken currentToken, StreamHandshakeToken handshakeToken)
         {
-            if (expectedToken != null)
+            if (this.expectedToken != null)
             {
-                if (!expectedToken.Equals(handshakeToken))
-                    return expectedToken;
+                if (!this.expectedToken.Equals(handshakeToken))
+                    return this.expectedToken;
             }
 
             await NextItem(item, currentToken);
 
             // check again, in case the expectedToken was changed indiretly via ResumeAsync()
-            if (expectedToken != null)
+            if (this.expectedToken != null)
             {
-                if (!expectedToken.Equals(handshakeToken))
-                    return expectedToken;
+                if (!this.expectedToken.Equals(handshakeToken))
+                    return this.expectedToken;
             }
             if (IsRewindable)
             {
-                expectedToken = StreamHandshakeToken.CreateDeliveyToken(currentToken);
+                this.expectedToken = StreamHandshakeToken.CreateDeliveyToken(currentToken);
             }
             return null;
         }
 
         public async Task NextBatch(IBatchContainerBatch batchContainerBatch)
         {
-            bool isRequestContextSet;
-            foreach (var batchContainer in batchContainerBatch.BatchContainers)
+            if (this.observer != null)
             {
-                isRequestContextSet = batchContainer.ImportRequestContext();
-                foreach (var itemTuple in batchContainer.GetEvents<T>())
-                {
-                    await NextItem(itemTuple.Item1, itemTuple.Item2);
-                }
 
-                if (isRequestContextSet)
```

Comment:
Was it intentional to remove the "RequestContext" support?

Reply:
Yes, but only in the case where one enabled batch delivery with a batched observer, so not in this case.  Will fix.